### PR TITLE
Update image layout properly

### PR DESCRIPTION
**Pull request by jaebaek**
Image layout of FrameBuffer for color attachment should have its layout as
`VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL` when we clear
or draw something on it. The one of depth/stencil attachment should have
`VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL`.

When we probe it, its layout must be
`VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL`. Note that we do not
change its layout to `VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL`
explicitly because the render pass will do.

Issue #7

### CODE REVIEW

File: src/vulkan/image.cc [hunk 1]
```diff
@@ -147,5 +147,82 @@ Result Image::CopyToHost(VkCommandBuffer command) {
...
+      break;
+    case VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL:
+      // An image becomes "transfer dst" only when we send a buffer data to
+      // it.
+      barrier.srcAccessMask = VK_ACCESS_SHADER_READ_BIT |
+                              VK_ACCESS_SHADER_WRITE_BIT |
+                              VK_ACCESS_TRANSFER_WRITE_BIT;
+      break;
+    default:
+      break;
```
Comment (dj2) [inline src/vulkan/image.cc]: Let's do barrier.srcAccessMask = 0; just to be clear what's happening.

File: src/vulkan/image.cc [hunk 2]
```diff
@@ -147,5 +147,82 @@ Result Image::CopyToHost(VkCommandBuffer command) {
...
+    case VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL:
+      // An image becomes "transfer dst" only when we send a buffer data to
+      // it.
+      barrier.dstAccessMask = VK_ACCESS_SHADER_READ_BIT |
+                              VK_ACCESS_SHADER_WRITE_BIT |
+                              VK_ACCESS_TRANSFER_WRITE_BIT;
+      break;
+    default:
+      break;
```
Comment (dj2) [inline src/vulkan/image.cc]: Same here barrier.dstAccessMask = 0;

File: src/vulkan/image.cc (hunk 1 above)
Comment (jaebaek) [reply src/vulkan/image.cc]: Done.

File: src/vulkan/image.cc (hunk 2 above)
Comment (jaebaek) [reply src/vulkan/image.cc]: Done.

**Pull request merged**